### PR TITLE
DBaaS Policy changes

### PR DESCRIPTION
**Pull request by mrrajan**
Signed-off-by: Rajan Ravi <rravi@redhat.com>

This PR includes changes for DBaaS policies, Existing script maintenance to avoid compile time error (works fine on run time) and new test scripts for DBaaS Policy error message

JIRA ticket: DBAAS-1012

Logs and Results - [result.zip…]()

### CODE REVIEW

File: resources/keywords/login.resource [hunk 1]
```diff
@@ -7,14 +7,15 @@ Library     ../../utils/scripts/dbaas_utils.py
 Library     ../../utils/scripts/util.py
 Resource    ../object_repo/login_obj.resource
 Resource    ../keywords/suite_and_test_teardown.resource
+Variables   ../../test-variables.yaml
```
Comment (ikanias) [inline resources/keywords/login.resource:10]: Why is that added?

File: resources/keywords/login.resource [hunk 2]
```diff
@@ -88,3 +90,23 @@ User Closes Welcome Message
     Run Keyword If    ${welcomeTile}
     ...    Click Element    ${btnWelcomTileClose}
 
+The User Logs In To Openshift As ${user}
```
Comment (ikanias) [inline resources/keywords/login.resource:93]: Where is this parameter comes from? Maybe the title of this keyword should be also changed to CLI?

File: resources/keywords/login.resource [hunk 3]
```diff
@@ -88,3 +90,23 @@ User Closes Welcome Message
...
+    Set Selenium Implicit Wait    5s
+    Open Browser    ${OCP_CONSOLE_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+    Select Login Authentication Type    ${${user}.AUTH_TYPE}
+    Login To Openshift As ${user}
+    Login To OpenShift CLI As LDAP User     ${${user}.USERNAME}     ${${user}.PASSWORD}     ${DBaaSTestNS}
```
Comment (ikanias) [inline resources/keywords/login.resource:98]: Why login to CLI appears here? Aren't we dealing with UI in this case?

File: resources/keywords/login.resource (hunk 1 above)
Comment (mrrajan) [reply resources/keywords/login.resource:10]: @ikanias Variables section included to avoid compile time errors, even without these the error would be resolved during runtime.

File: resources/keywords/login.resource (hunk 2 above)
Comment (mrrajan) [reply resources/keywords/login.resource:93]: @ikanias This keyword includes the user logs in with UI as well as CLI and this parameter filled out while calling the function like "The User Logs In To Openshift As OCP_LDAP_PROJECT_ADM"

File: resources/keywords/login.resource (hunk 3 above)
Comment (mrrajan) [reply resources/keywords/login.resource:98]: We are using CLI as well as UI to perform operations like creating service binding which can be only performed with CLI.

File: resources/keywords/suite_and_test_teardown.resource [hunk 4]
```diff
@@ -70,3 +71,33 @@ Tear Down The Test Suite
     ...    AND    Setup DBaaSPolicy For ${operatorNamespace}
```
Comment (ikanias) [inline resources/keywords/suite_and_test_teardown.resource:71]: Don't really understand why all this is under teardown

File: resources/keywords/suite_and_test_teardown.resource [hunk 5]
```diff
@@ -70,3 +71,33 @@ Tear Down The Test Suite
...
     ...    AND    Set Global Variable    ${DBaaSPolicyConfigured}    ${True}
 
+Configure DBaaS Policy Namespaces
```
Comment (ikanias) [inline resources/keywords/suite_and_test_teardown.resource:85]: Don't understand why all of this is under teardown. It looks like it is supposed to be in the test setup or login

File: test-variables.yaml.example [hunk 6]
```diff
@@ -13,6 +13,18 @@ OCP_LDAP_USER:
   AUTH_TYPE: "ldap-provider-qe"
   USERNAME: ldap_user
   PASSWORD: abc123XYZ!
+OCP_LDAP_PROJECT_ADM:
+  AUTH_TYPE: "ldap-provider-qe"
+  USERNAME: ldap_user
+  PASSWORD: abc123XYZ!
+OCP_LDAP_SERVICE_ADM:
+  AUTH_TYPE: "ldap-provider-qe"
+  USERNAME: ldap_user
```
Comment (ikanias) [inline test-variables.yaml.example:22]: Why do all the personas have the same username?

File: tests/dbaas_policy.robot [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+Library             SeleniumLibrary
+Resource            ../resources/keywords/dbaas_policy.resource
+
+Suite Setup         Configure DBaaS Policy Namespaces
+Suite Teardown      Tear Down The Test Suite
+Test Teardown       Close Browser
+Force Tags          UI  DBaaS
+
+*** Test Cases ***
+Scenario: Verify Invalid Namespace Error Message For Project Admin To Create DBaaS Policy
```
Comment (ikanias) [inline tests/dbaas_policy.robot:14]: I assume project admin cannot create DBaaS Policy right?

File: utils/scripts/testconfig/test-variables.yaml [hunk 8]
```diff
@@ -12,6 +12,18 @@ OCP_LDAP_USER:
   AUTH_TYPE: OCP_LDAP_USER_AUTH_TYPE
   USERNAME: OCP_LDAP_USER_USERNAME
   PASSWORD: OCP_LDAP_USER_PASSWORD
+OCP_LDAP_PROJECT_ADM:
```
Comment (ikanias) [inline utils/scripts/testconfig/test-variables.yaml:15]: Maybe in order to clear the code and make it more understandable, we should change the OCP_LDAP_USER to be OCP_LDAP_CLUSTER_ADM? Do we refer to a cluster admin there?

File: test-variables.yaml.example (hunk 6 above)
Comment (mrrajan) [reply test-variables.yaml.example:22]: @ikanias These fields needs to be updated by the user to represent the actuals

File: tests/dbaas_policy.robot (hunk 7 above)
Comment (mrrajan) [reply tests/dbaas_policy.robot:14]: No, They could. But when there is no dbaas policy available, then there should be an error message while creating DBaaSInventory. So this test is to verify the error message.

File: utils/scripts/testconfig/test-variables.yaml (hunk 8 above)
Comment (mrrajan) [reply utils/scripts/testconfig/test-variables.yaml:15]: These fields are just for reference and will be updated during runtime on jenkins or by the user while execution. IMHO, this is fine. Please let me know if it still needs to be changed.

File: resources/keywords/suite_and_test_teardown.resource (hunk 5 above)
Comment (mrrajan) [reply resources/keywords/suite_and_test_teardown.resource:85]: Thanks for pointing it out, I planned to refactor the resource file name and missed it out. Updated resource file name from "resources/keywords/suite_and_test_teardown.resource" to "resources/keywords/suite_and_test_setup_teardown.resource"

**Review by ikanias — APPROVED** [review]
LGTM

**Review by harshkumarRH — APPROVED** [review]
Looks good to me
